### PR TITLE
MULE-11065: Make all configuration components AnnotatedObject

### PR DESCRIPTION
**Pull request by elrodro83**
* Make MessageRedeliveredException not extends MessagingException
* Allow to dynamically make an object `annotated` through CGLib

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/util/ExceptionUtils.java [hunk 1]
```diff
@@ -217,19 +216,15 @@ public static String getFullStackTraceWithoutMessages(Throwable throwable) {
...
         exception instanceof WrapperErrorMessageAwareException ? ((WrapperErrorMessageAwareException) exception).getRootCause()
             : exception;
-    ComponentIdentifier componentIdentifier = null;
-    List<ErrorMapping> errorMappings = null;
-    if (AnnotatedObject.class.isAssignableFrom(annotatedObject.getClass())) {
-      componentIdentifier =
-          (ComponentIdentifier) ((AnnotatedObject) annotatedObject).getAnnotation(ANNOTATION_NAME);
-      errorMappings = (List<ErrorMapping>) ((AnnotatedObject) annotatedObject).getAnnotation(ANNOTATION_ERROR_MAPPINGS);
-    }
+    ComponentIdentifier componentIdentifier = (ComponentIdentifier) annotatedObject.getAnnotation(ANNOTATION_NAME);
```
Comment (pablolagreca) [inline core/src/main/java/org/mule/runtime/core/api/util/ExceptionUtils.java]: This should be annotatedObject.getLocation().getIdentifier()

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/LifecycleAwareConfigurationInstance.java [hunk 2]
```diff
@@ -187,22 +188,26 @@ private void testConnectivity() throws MuleException {
...
-        if (lockAcquired) {
-          LOGGER.info("Doing testConnectivity() for config " + getName());
-          try {
-            ConnectionValidationResult result = connectionManager.testConnectivity(LifecycleAwareConfigurationInstance.this);
-            if (result.isValid()) {
-              context.setOk();
-            } else {
-              context.setFailed(result.getException());
-              throw new ConnectionException(format("Connectivity test failed for config '%s'", getName()), result.getException());
+        Lock lock = testConnectivityLock;
```
Comment (pablolagreca) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/LifecycleAwareConfigurationInstance.java]: Is this change related to the feature?

Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/LifecycleAwareConfigurationInstance.java]: no, but it caused some NPE to appear on the log that annoyed me
